// @openauthjs/openauth/dist/esm/ui/base.js
// src/ui/ui.css
var ui_default = `@import url("https://unpkg.com/tailwindcss@3.4.15/src/css/preflight.css");

:root {
  --color-background-dark: #0e0e11;
  --color-background-light: #ffffff;
  --color-primary-dark: #6772e5;
  --color-primary-light: #6772e5;
  --border-radius: 0;

  --color-background: var(--color-background-dark);
  --color-primary: var(--color-primary-dark);

  @media (prefers-color-scheme: light) {
    --color-background: var(--color-background-light);
    --color-primary: var(--color-primary-light);
  }

  --color-high: oklch(
    from var(--color-background) calc(round(1 - round(l / 1.4))) 0 0
  );
  --color-low: oklch(from var(--color-background) calc(round(l / 1.4)) 0 0);
  --lightness-high: color-mix(
    in oklch,
    var(--color-high) 0%,
    oklch(var(--color-high) 0 0)
  );
  --lightness-low: color-mix(
    in oklch,
    var(--color-low) 0%,
    oklch(var(--color-low) 0 0)
  );
  --font-family: ui-sans-serif, system-ui, sans-serif, "Apple Color Emoji",
    "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
  --font-scale: 1;

  --font-size-xs: calc(0.75rem * var(--font-scale));
  --font-size-sm: calc(0.875rem * var(--font-scale));
  --font-size-md: calc(1rem * var(--font-scale));
  --font-size-lg: calc(1.125rem * var(--font-scale));
  --font-size-xl: calc(1.25rem * var(--font-scale));
  --font-size-2xl: calc(1.5rem * var(--font-scale));
}

[data-component="root"] {
  font-family: var(--font-family);
  background-color: var(--color-background);
  padding: 1rem;
  color: white;
  position: absolute;
  inset: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  user-select: none;
  color: var(--color-high);
}

[data-component="center"] {
  width: 380px;
  display: flex;
  flex-direction: column;
  gap: 1.5rem;

  &[data-size="small"] {
    width: 300px;
  }
}

[data-component="link"] {
  text-decoration: underline;
  font-weight: 600;
}

[data-component="label"] {
  display: flex;
  gap: 0.75rem;
  flex-direction: column;
  font-size: var(--font-size-xs);
}

[data-component="logo"] {
  margin: 0 auto;
  height: 2.5rem;
  width: auto;
  display: none;

  @media (prefers-color-scheme: light) {
    &[data-mode="light"] {
      display: block;
    }
  }

  @media (prefers-color-scheme: dark) {
    &[data-mode="dark"] {
      display: block;
    }
  }
}

[data-component="input"] {
  width: 100%;
  height: 2.5rem;
  padding: 0 1rem;
  border: 1px solid transparent;
  --background: oklch(
    from var(--color-background) calc(l + (-0.06 * round(l / 1.4) + 0.03)) c h
  );
  background: var(--background);
  border-color: oklch(
    from var(--color-background)
      calc(clamp(0.22, l + (-0.12 * round(l / 1.4) + 0.06), 0.88)) c h
  );
  border-radius: calc(var(--border-radius) * 0.25rem);
  font-size: var(--font-size-sm);
  outline: none;

  &:focus {
    border-color: oklch(
      from var(--color-background)
        calc(clamp(0.3, l + (-0.2 * round(l / 1.4) + 0.1), 0.7)) c h
    );
  }

  &:user-invalid:not(:focus) {
    border-color: oklch(0.4 0.09 7.91);
  }
}

[data-component="button"] {
  height: 2.5rem;
  cursor: pointer;
  border: 0;
  font-weight: 500;
  font-size: var(--font-size-sm);
  border-radius: calc(var(--border-radius) * 0.25rem);
  display: flex;
  gap: 0.75rem;
  align-items: center;
  justify-content: center;
  background: var(--color-primary);
  color: oklch(from var(--color-primary) calc(round(1 - round(l / 1.4))) 0 0);

  &[data-color="ghost"] {
    background: transparent;
    color: var(--color-high);
    border: 1px solid
      oklch(
        from var(--color-background)
          calc(clamp(0.22, l + (-0.12 * round(l / 1.4) + 0.06), 0.88)) c h
      );
  }

  [data-slot="icon"] {
    width: 16px;
    height: 16px;

    svg {
      width: 100%;
      height: 100%;
    }
  }
}

[data-component="form"] {
  max-width: 100%;
  display: flex;
  flex-direction: column;
  gap: 1rem;
  margin: 0;
}

[data-component="form-alert"] {
  height: 2.5rem;
  display: flex;
  align-items: center;
  padding: 0 1rem;
  border-radius: calc(var(--border-radius) * 0.25rem);
  background: oklch(0.28 0.06 7.91);
  color: oklch(0.78 0.16 22.29);
  text-align: left;
  font-size: 0.75rem;
  gap: 0.5rem;

  &[data-color="success"] {
    background: oklch(0.28 0.06 149.45);
    color: oklch(0.78 0.16 162.29);
  }

  &:has([data-slot="message"]:empty) {
    display: none;
  }

  [data-slot="icon"] {
    width: 1rem;
    height: 1rem;
  }
}

[data-component="form-footer"] {
  display: flex;
  gap: 1rem;
  font-size: 0.75rem;
  align-items: center;
  justify-content: center;

  &:has(> :nth-child(2)) {
    justify-content: space-between;
  }
}
`;

// src/ui/theme.ts
var THEME_SST = {
  title: "SST",
  logo: {
    dark: "https://sst.dev/favicon.svg",
    light: "https://sst.dev/favicon.svg"
  },
  background: {
    dark: "#1a1a2d",
    light: "rgb(255, 255, 255)"
  },
  primary: "#f3663f",
  font: {
    family: "Rubik, sans-serif"
  },
  css: `
    @import url('https://fonts.googleapis.com/css2?family=Rubik:wght@100;200;300;400;500;600;700;800;900&display=swap');
  `
};
function getTheme() {
  return globalThis.OPENAUTH_THEME || THEME_SST;
}

// src/ui/base.tsx
import { jsxDEV } from "hono/jsx/jsx-dev-runtime";
function Layout(props) {
  const theme = getTheme();
  function get(key, mode) {
    if (!theme)
      return;
    if (!theme[key])
      return;
    if (typeof theme[key] === "string")
      return theme[key];
    return theme[key][mode];
  }
  const radius = (() => {
    if (theme?.radius === "none")
      return "0";
    if (theme?.radius === "sm")
      return "1";
    if (theme?.radius === "md")
      return "1.25";
    if (theme?.radius === "lg")
      return "1.5";
    if (theme?.radius === "full")
      return "1000000000001";
    return "1";
  })();
  return /* @__PURE__ */ jsxDEV("html", {
    style: {
      "--color-background-light": get("background", "light"),
      "--color-background-dark": get("background", "dark"),
      "--color-primary-light": get("primary", "light"),
      "--color-primary-dark": get("primary", "dark"),
      "--font-family": theme?.font?.family,
      "--font-scale": theme?.font?.scale,
      "--border-radius": radius
    },
    children: [
      /* @__PURE__ */ jsxDEV("head", {
        children: [
          /* @__PURE__ */ jsxDEV("title", {
            children: theme?.title || "OpenAuthJS"
          }, undefined, false, undefined, this),
          /* @__PURE__ */ jsxDEV("meta", {
            name: "viewport",
            content: "width=device-width, initial-scale=1"
          }, undefined, false, undefined, this),
          /* @__PURE__ */ jsxDEV("link", {
            rel: "icon",
            href: theme?.favicon
          }, undefined, false, undefined, this),
          /* @__PURE__ */ jsxDEV("style", {
            dangerouslySetInnerHTML: { __html: ui_default }
          }, undefined, false, undefined, this),
          theme?.css && /* @__PURE__ */ jsxDEV("style", {
            dangerouslySetInnerHTML: { __html: theme.css }
          }, undefined, false, undefined, this)
        ]
      }, undefined, true, undefined, this),
      /* @__PURE__ */ jsxDEV("body", {
        children: /* @__PURE__ */ jsxDEV("div", {
          "data-component": "root",
          children: /* @__PURE__ */ jsxDEV("div", {
            "data-component": "center",
            "data-size": props.size,
            children: [
              /* @__PURE__ */ jsxDEV("img", {
                "data-component": "logo",
                src: get("logo", "light"),
                "data-mode": "light"
              }, undefined, false, undefined, this),
              /* @__PURE__ */ jsxDEV("img", {
                "data-component": "logo",
                src: get("logo", "dark"),
                "data-mode": "dark"
              }, undefined, false, undefined, this),
              props.children
            ]
          }, undefined, true, undefined, this)
        }, undefined, false, undefined, this)
      }, undefined, false, undefined, this)
    ]
  }, undefined, true, undefined, this);
}
export {
  Layout
};
