// @opencode-ai/sdk/dist/gen/sdk.gen.js
// This file is auto-generated by @hey-api/openapi-ts
import { client as _heyApiClient } from "./client.gen.js";
class _HeyApiClient {
    _client = _heyApiClient;
    constructor(args) {
        if (args?.client) {
            this._client = args.client;
        }
    }
}
class Event extends _HeyApiClient {
    /**
     * Get events
     */
    subscribe(options) {
        return (options?.client ?? this._client).get({
            url: "/event",
            ...options,
        });
    }
}
class App extends _HeyApiClient {
    /**
     * Get app info
     */
    get(options) {
        return (options?.client ?? this._client).get({
            url: "/app",
            ...options,
        });
    }
    /**
     * Initialize the app
     */
    init(options) {
        return (options?.client ?? this._client).post({
            url: "/app/init",
            ...options,
        });
    }
    /**
     * Write a log entry to the server logs
     */
    log(options) {
        return (options?.client ?? this._client).post({
            url: "/log",
            ...options,
            headers: {
                "Content-Type": "application/json",
                ...options?.headers,
            },
        });
    }
    /**
     * List all agents
     */
    agents(options) {
        return (options?.client ?? this._client).get({
            url: "/agent",
            ...options,
        });
    }
}
class Config extends _HeyApiClient {
    /**
     * Get config info
     */
    get(options) {
        return (options?.client ?? this._client).get({
            url: "/config",
            ...options,
        });
    }
    /**
     * List all providers
     */
    providers(options) {
        return (options?.client ?? this._client).get({
            url: "/config/providers",
            ...options,
        });
    }
}
class Session extends _HeyApiClient {
    /**
     * List all sessions
     */
    list(options) {
        return (options?.client ?? this._client).get({
            url: "/session",
            ...options,
        });
    }
    /**
     * Create a new session
     */
    create(options) {
        return (options?.client ?? this._client).post({
            url: "/session",
            ...options,
        });
    }
    /**
     * Delete a session and all its data
     */
    delete(options) {
        return (options.client ?? this._client).delete({
            url: "/session/{id}",
            ...options,
        });
    }
    /**
     * Get session
     */
    get(options) {
        return (options.client ?? this._client).get({
            url: "/session/{id}",
            ...options,
        });
    }
    /**
     * Analyze the app and create an AGENTS.md file
     */
    init(options) {
        return (options.client ?? this._client).post({
            url: "/session/{id}/init",
            ...options,
            headers: {
                "Content-Type": "application/json",
                ...options.headers,
            },
        });
    }
    /**
     * Abort a session
     */
    abort(options) {
        return (options.client ?? this._client).post({
            url: "/session/{id}/abort",
            ...options,
        });
    }
    /**
     * Unshare the session
     */
    unshare(options) {
        return (options.client ?? this._client).delete({
            url: "/session/{id}/share",
            ...options,
        });
    }
    /**
     * Share a session
     */
    share(options) {
        return (options.client ?? this._client).post({
            url: "/session/{id}/share",
            ...options,
        });
    }
    /**
     * Summarize the session
     */
    summarize(options) {
        return (options.client ?? this._client).post({
            url: "/session/{id}/summarize",
            ...options,
            headers: {
                "Content-Type": "application/json",
                ...options.headers,
            },
        });
    }
    /**
     * List messages for a session
     */
    messages(options) {
        return (options.client ?? this._client).get({
            url: "/session/{id}/message",
            ...options,
        });
    }
    /**
     * Create and send a new message to a session
     */
    chat(options) {
        return (options.client ?? this._client).post({
            url: "/session/{id}/message",
            ...options,
            headers: {
                "Content-Type": "application/json",
                ...options.headers,
            },
        });
    }
    /**
     * Get a message from a session
     */
    message(options) {
        return (options.client ?? this._client).get({
            url: "/session/{id}/message/{messageID}",
            ...options,
        });
    }
    /**
     * Revert a message
     */
    revert(options) {
        return (options.client ?? this._client).post({
            url: "/session/{id}/revert",
            ...options,
            headers: {
                "Content-Type": "application/json",
                ...options.headers,
            },
        });
    }
    /**
     * Restore all reverted messages
     */
    unrevert(options) {
        return (options.client ?? this._client).post({
            url: "/session/{id}/unrevert",
            ...options,
        });
    }
}
class Find extends _HeyApiClient {
    /**
     * Find text in files
     */
    text(options) {
        return (options.client ?? this._client).get({
            url: "/find",
            ...options,
        });
    }
    /**
     * Find files
     */
    files(options) {
        return (options.client ?? this._client).get({
            url: "/find/file",
            ...options,
        });
    }
    /**
     * Find workspace symbols
     */
    symbols(options) {
        return (options.client ?? this._client).get({
            url: "/find/symbol",
            ...options,
        });
    }
}
class File extends _HeyApiClient {
    /**
     * Read a file
     */
    read(options) {
        return (options.client ?? this._client).get({
            url: "/file",
            ...options,
        });
    }
    /**
     * Get file status
     */
    status(options) {
        return (options?.client ?? this._client).get({
            url: "/file/status",
            ...options,
        });
    }
}
class Tui extends _HeyApiClient {
    /**
     * Append prompt to the TUI
     */
    appendPrompt(options) {
        return (options?.client ?? this._client).post({
            url: "/tui/append-prompt",
            ...options,
            headers: {
                "Content-Type": "application/json",
                ...options?.headers,
            },
        });
    }
    /**
     * Open the help dialog
     */
    openHelp(options) {
        return (options?.client ?? this._client).post({
            url: "/tui/open-help",
            ...options,
        });
    }
    /**
     * Open the session dialog
     */
    openSessions(options) {
        return (options?.client ?? this._client).post({
            url: "/tui/open-sessions",
            ...options,
        });
    }
    /**
     * Open the theme dialog
     */
    openThemes(options) {
        return (options?.client ?? this._client).post({
            url: "/tui/open-themes",
            ...options,
        });
    }
    /**
     * Open the model dialog
     */
    openModels(options) {
        return (options?.client ?? this._client).post({
            url: "/tui/open-models",
            ...options,
        });
    }
    /**
     * Submit the prompt
     */
    submitPrompt(options) {
        return (options?.client ?? this._client).post({
            url: "/tui/submit-prompt",
            ...options,
        });
    }
    /**
     * Clear the prompt
     */
    clearPrompt(options) {
        return (options?.client ?? this._client).post({
            url: "/tui/clear-prompt",
            ...options,
        });
    }
    /**
     * Execute a TUI command (e.g. switch_agent)
     */
    executeCommand(options) {
        return (options?.client ?? this._client).post({
            url: "/tui/execute-command",
            ...options,
            headers: {
                "Content-Type": "application/json",
                ...options?.headers,
            },
        });
    }
}
export class OpencodeClient extends _HeyApiClient {
    /**
     * Respond to a permission request
     */
    postSessionByIdPermissionsByPermissionId(options) {
        return (options.client ?? this._client).post({
            url: "/session/{id}/permissions/{permissionID}",
            ...options,
            headers: {
                "Content-Type": "application/json",
                ...options.headers,
            },
        });
    }
    event = new Event({ client: this._client });
    app = new App({ client: this._client });
    config = new Config({ client: this._client });
    session = new Session({ client: this._client });
    find = new Find({ client: this._client });
    file = new File({ client: this._client });
    tui = new Tui({ client: this._client });
}
