// @openauthjs/openauth/dist/esm/keys.js
// src/keys.ts
import {
  exportJWK,
  exportPKCS8,
  exportSPKI,
  generateKeyPair,
  importPKCS8,
  importSPKI
} from "jose";
import { Storage } from "./storage/storage.js";
var alg = "RS512";
async function keys(storage) {
  const results = [];
  const scanner = Storage.scan(storage, ["oauth:key"]);
  for await (const [_key, value] of scanner) {
    const publicKey = await importSPKI(value.publicKey, alg, {
      extractable: true
    });
    const privateKey = await importPKCS8(value.privateKey, alg);
    const jwk = await exportJWK(publicKey);
    jwk.kid = value.id;
    results.push({
      id: value.id,
      alg,
      created: new Date(value.created),
      signing: {
        public: publicKey,
        private: privateKey
      },
      encryption: {
        public: await importSPKI(value.publicKey, "RSA-OAEP-512"),
        private: await importPKCS8(value.privateKey, "RSA-OAEP-512")
      },
      jwk
    });
  }
  if (results.length)
    return results;
  const key = await generateKeyPair(alg, {
    extractable: true
  });
  const serialized = {
    id: crypto.randomUUID(),
    publicKey: await exportSPKI(key.publicKey),
    privateKey: await exportPKCS8(key.privateKey),
    created: Date.now()
  };
  await Storage.set(storage, ["oauth:key", serialized.id], serialized);
  return keys(storage);
}
export {
  keys
};
